examples/correlation_alignment_example.py

```diff
@@ -0,0 +1,84 @@
+"""Align data with significant frequency drift
+==============================================
+
+Takes a 2D data set and applies proper phasing corrections followed by 
+aligning the data through a correlation routine.
+"""
+from pyspecdata import *
+from pyspecProcScripts import *
+from pylab import *
+import sympy as s
+from collections import OrderedDict
+from numpy.random import normal,seed
+seed(2021)
+rcParams['image.aspect'] = 'auto' #needed for sphinx gallery
+
+#{{{generate fake data
+fl = figlist_var()
+#this generates fake clean data with a T2 of 0.2s
+# amplitude of 23 (arbitrary), offset of 300 Hz,
+# FWHM 10 Hz
+f_range = (-400,400)
+t2, td, vd, ph1, ph2 = s.symbols('t2 td vd ph1 ph2')
+echo_time = 1e-3
+data = fake_data(
+        23*(1 - 2*s.exp(-vd / 0.2)) * s.exp(+1j*2*s.pi*100*(t2) - abs(t2)*50*s.pi),
```

Comment:
I would put everything inside a loop, then loop over two functions (1) this -- the inv. rec. (2) an enhancement curve

Reply:
I'm confused... loop to make multiple datasets?

Reply:
and I am running into trouble running this now because at this point in the master branch fake_data is not a function

Reply:
yes -- this way you will demo the correlation on both an inversion recovery and an enhancement curve.

Reply:
actually, if you just pull this branch now, it will work -- I think your master is out of date -- when I merged in an up to date master, what you had worked fine for me.

Reply:
if you want to update master without actually being on master (e.g. if you're working on the correlation_alignment branch), you can usually do `git fetch origin master:master` (you can tab complete the last part).

Reply:
I'm having a hard time visualizing how to do the loop within a loop... I first tried using lists and having a list for the datasets, f_ranges, and signal pathways but ran into trouble with unpacking the list of dictionaries for the signal pathways....

Reply:
I'm not sure why you would have a loop within a loop.  What you describe here makes sense .. probably just a syntax issue?

Reply:
hmm, I realized this is likely an old comment...